sqldb/sqlite.go

```diff
@@ -165,3 +165,29 @@ func NewTestSqliteDB(t *testing.T) *SqliteStore {
 
 	return sqlDB
 }
+
+// NewTestSqliteDBWithVersion is a helper function that creates an SQLite
+// database for testing and migrates it to the given version.
+func NewTestSqliteDBWithVersion(t *testing.T, version uint) *SqliteStore {
+	t.Helper()
+
+	t.Logf("Creating new SQLite DB for testing, migrating to version %d",
+		version)
+
+	// TODO(roasbeef): if we pass :memory: for the file name, then we get
+	// an in mem version to speed up tests
+	dbFileName := filepath.Join(t.TempDir(), "tmp.db")
+	sqlDB, err := NewSqliteStore(&SqliteConfig{
+		SkipMigrations: true,
```

Comment:
Given that this is for testing only imo it's better to be explicit since functional options also require developers to look up usage in the code and compiler won't help catching incorrect use. 

I agree though it's a nice way to hide complexity :) Happy to change it if you feel it's better!